gh-132470: Prevent crash in ctypes.CField when `byte_size` is incorrect (#132475)
python/cpython: Misc/NEWS.d/next/C_API/2025-04-13-20-52-39.gh-issue-132470.UqBQjN.rst

Reviewer: What on earth happened to your news entry? Just redo this using `blurb_it`, please.

Reviewer 2: @ZeroIntensity We can always suggest an impovment here on github :)

Many new contributors might not know how to work with cpython-specific tooling and what NEWS even are.

Here's my suggestion: https://github.com/python/cpython/pull/132475#discussion_r2046355923

Reviewer: Thanks for that, I didn't mean to come off as rude :). In hindsight, I didn't see the "first time contributor" mark, so that's my bad!

> Creating a ctypes.CField with a byte_size that does not match the actual
> type size now raises a ValueError instead of crashing the interpreter. #
> Please enter the relevant GitHub issue number here: #
> 
> # # Uncomment one of these "section:" lines to specify which section # this
> entry should go in in Misc/NEWS.d. # #.. section: Security #.. section: Core
> and Builtins #.. section: Library #.. section: Documentation #.. section:
> Tests #.. section: Build #.. section: Windows #.. section: macOS #..
> section: IDLE #.. section: Tools/Demos #.. section: C API
> 
> # Write your Misc/NEWS.d entry below.  It should be a simple ReST paragraph.
> # Don't start with "- Issue #<n>: " or "- gh-issue-<n>: " or that sort of
> stuff.
> ###########################################################################

Reviewer 2: ```suggestion
Creating a :class:`ctypes.CField` with a *byte_size* that does not match the actual
type size now raises a :exc:`ValueError` instead of crashing the interpreter.
```
